Occasional panic when redeploying module

```
info:tickerdb: Found file changes: +db/queries/stockdb/queries.sql
info:tickerdb:build: Building module
info:tickerdb:build: Module built (1.81s)
info:tickerdb: API server listening at: 127.0.0.1:52729
info:build-engine: Changeset chs-1du4hvd02gw3vda4 deployed and ready
✔[ticker]      ✔[tickerdb]    ◞[🔮 Goose]  
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x103b94d60]

goroutine 3648 [running]:
os.(*ProcessState).exited(...)
	/Users/mtoohey/Library/Caches/hermit/pkg/go-1.23.6/src/os/exec_posix.go:93
os.(*ProcessState).Exited(...)
	/Users/mtoohey/Library/Caches/hermit/pkg/go-1.23.6/src/os/exec.go:381
github.com/block/ftl/backend/runner.(*Service).Close(0x14002c52908)
	/Users/mtoohey/Code/ftl/backend/runner/runner.go:524 +0x2f0
github.com/block/ftl/backend/runner.(*Service).deploy.func3()
	/Users/mtoohey/Code/ftl/backend/runner/runner.go:481 +0x30
created by context.(*afterFuncCtx).cancel.func1 in goroutine 3730
	/Users/mtoohey/Library/Caches/hermit/pkg/go-1.23.6/src/context/context.go:348 +0x24
```

--- a/backend/runner/runner.go
+++ b/backend/runner/runner.go
@@ -521,7 +521,7 @@ func (s *Service) Close() error {
 		case <-done:
 		case <-depl.ctx.Done():
 		case <-time.After(10 * time.Second):
-			if !cmd.ProcessState.Exited() {
+			if cmd.ProcessState == nil || !cmd.ProcessState.Exited() {
 				err := cmd.Kill(syscall.SIGKILL)
 				if err != nil {
 					return fmt.Errorf("failed to kill plugin: %w", err)

